pkg/sharings/sharings.go

```diff
@@ -106,15 +106,53 @@ func GetRecipient(db couchdb.Database, recID string) (*Recipient, error) {
 	return doc, err
 }
 
-// CheckSharingCreation initializes and check some sharing fields at creation
-func CheckSharingCreation(db couchdb.Database, sharing *Sharing) error {
-
-	sharingType := sharing.SharingType
+//CheckSharingType returns an error if the sharing type is incorrect
+func CheckSharingType(sharingType string) error {
 	if sharingType != consts.OneShotSharing &&
 		sharingType != consts.MasterSlaveSharing &&
 		sharingType != consts.MasterMasterSharing {
 		return ErrBadSharingType
 	}
+	return nil
```

Comment:
FYI, it can also be written with a switch case, like this:

```go
switch sharingType {
case consts.OneShotSharing, consts.MasterSlaveSharing, consts.MasterMasterSharing:
    return nil
}
return ErrBadSharingType
```